File: demo/src/main/java/com/example/demo/ui/auth/token/TokenRepository.java
package com.example.demo.ui.auth.token;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.List;
import java.util.Optional;

public interface TokenRepository extends JpaRepository<TokenEntity, Integer> {

  @Query(value = """
      select t from TokenEntity t inner join UserEntity u
      on t.user.id = u.id
      where u.name = :name and (t.expired = false or t.revoked = false)
      """)
  List<TokenEntity> findAllValidTokenByUser(String name);

  Optional<TokenEntity> findByToken(String token);
}

File: demoKotlin/demoKotlin/backend/src/main/kotlin/org/example/backend/data/repositories/TokenRepository.kt
package org.example.backend.data.repositories

import org.example.backend.data.modelo.TokenEntity
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.ListCrudRepository
import org.springframework.stereotype.Repository

@Repository
interface TokenRepository : ListCrudRepository<TokenEntity, Long> {

    @Query("""
      select t from TokenEntity t inner join UserEntity u
      on t.user.id = u.id
      where u.name = :name and (t.expired = false or t.revoked = false)
      
      """)
    fun findAllValidTokenByUser(name: String?): List<TokenEntity>
    fun findByToken(jwt: String): TokenEntity?

}
